remix/app/services/auth.server.ts

```diff
@@ -0,0 +1,11 @@
+import type { UserProfile } from "~/types";
+import UserRespository from "~/repositories/user.server";
+
+export async function handleSocialAuthCallBack({ profile }: any) {
+  const userProfile: UserProfile = profile?._json;
+
+  if (userProfile) {
+    await UserRespository.updateOrCreate(userProfile)
+  }
```

Comment:
Add also a line break before the return statements. Always consider breaking code into: variables, logic, and return values.